### ISSUE TITLE
Distribution value has |bucket_counts| fields that sum to 4 which does not equal the |count| field value of 2

### ISSUE THREAD

**Issue by altunkan** [micrometer-metrics/micrometer#4868]
**Describe the bug**
Hello,

Recently we have migrated our Spring Boot application from 2.7 to 3.2 and we've started to observe following error.

```
com.google.api.gax.rpc.InvalidArgumentException: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: One or more TimeSeries could not be written: Field timeSeries[31].points[0].distributionValue had an invalid value: Distribution value has |bucket_counts| fields that sum to 4 which does not equal the |count| field value of 2.
```

**Environment**

 - Micrometer version: 1.12.2
 - Micrometer registry: Stackdriver
 - OS: Linux
 - Java version: 17

**To Reproduce**
How to reproduce the bug:
```
  private Timer buildDependencyTimer(DependencyTimer dependencyTimer) {
    return Timer.builder(LATENCY_DEPENDENCY_PREFIX)
        .tags(getDependencyTags(dependencyTimer))
        .publishPercentiles(percentiles)
        .publishPercentileHistogram()
        .minimumExpectedValue(Duration.ofNanos(1L))
        .register(meterRegistry);
  }

    dependencyLatencyTimers.putIfAbsent(dependencyTimer, buildDependencyTimer(dependencyTimer));
    dependencyLatencyTimers.get(dependencyTimer).record(responseTime, TimeUnit.MILLISECONDS);
```

**Expected behavior**
Metrics should be reported to stackdriver.

**Additional context**
Add any other context about the problem here, e.g. related issues.

**Comment by hudson155** [comment]
Seeing the same thing on a new project @altunkan is this new after a version upgrade? I might try older verisons...

**Comment by hudson155** [comment]
I got it working by downgrading to 1.11.6.

**Comment by altunkan** [comment]
Yeah, I just tried with `1.11.6` by adding `set('micrometer.version', '1.11.6')` and looks like error is gone. Thank you! This will keep us good until micrometer team addresses the issue.

**Comment by rafal-dudek** [comment]
I believe this is the same issue as that: https://github.com/micrometer-metrics/micrometer/issues/998 
The problem is there since 2018 but looks like on Spring Boot 3.2 it is much more common.
We had a sporadic problem with it in our company since 2021, but after migration to Spring Boot 3.2 we have already dozens of applications experiencing it regularly.

**Comment by altunkan** [comment]
Unfortunately bug still happens with version `1.11.6`. 

All of our count based metrics fluctuate and we disabled error logging for that specific class.

**Comment by shakuzen** [comment]
I'm hoping #5836 fixes this issue. I did some testing with it and had positive results, but it would be much appreciated if users could verify the fix once it is available in snapshots. I will leave another comment here when that is.

**Comment by shakuzen** [comment]
The fix for this is now available in snapshots for 1.13.11, 1.14.4, and 1.15.0. If anyone affected by this could test out the fix in snapshots, it would be very appreciated.

**Comment by shakuzen** [comment]
This is now released in the versions mentioned in my previous comment. We would still be grateful for some feedback on if it solves the issue for everyone affected.

**Comment by gruffT** [comment]
I can confirm we saw a significant volume of these messages stop with migration to `1.14.4`. Thank you!
